superset/dashboards/dao.py

```diff
@@ -37,40 +36,20 @@
 
 class DashboardDAO(BaseDAO):
     model_cls = Dashboard
-    base_filter = DashboardFilter
+    base_filter = DashboardAccessFilter
 
     @staticmethod
     def get_by_id_or_slug(id_or_slug: str) -> Dashboard:
-        query = (
-            db.session.query(Dashboard)
-            .filter(id_or_slug_filter(id_or_slug))
-            .outerjoin(Slice, Dashboard.slices)
-            .outerjoin(Slice.table)
-            .outerjoin(Dashboard.owners)
-            .outerjoin(Dashboard.roles)
-        )
-        # Apply dashboard base filters
-        query = DashboardFilter("id", SQLAInterface(Dashboard, db.session)).apply(
-            query, None
-        )
-        dashboard = query.one_or_none()
+        dashboard = Dashboard.get(id_or_slug)
         if not dashboard:
             raise DashboardNotFoundError()
```

Comment:
this exposes a new security risk: it would be easy to know if a dashboard exists or not even if user doesn't have access to begin with

Reply:
But that's just the same as before right? The goal of this PR is the revert the behavior of non-owners having access to draft dashboards. Whether this is a risk that worth further actions is another topic.